client/my-sites/design-preview/index.js

```diff
@@ -165,36 +168,43 @@ const DesignPreview = React.createClass( {
 			return;
 		}
 		event.preventDefault();
-		// TODO: if the href is on the current site, load the href as a preview and fetch markup for that url
 	},
 
-	getPreviewUrl( site ) {
-		if ( ! site ) {
+	getPreviewUrl() {
+		if ( ! this.props.selectedSiteUrl && ! this.props.previewUrl ) {
```

Comment:
TIL again!

Why don't we just fix that "empty query arg" issue in `addQueryArgs`? Added a commit. Let me know what you think.